src/client/src/components/TasksList/TasksList.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import TaskItem from "../TaskItem/TaskItem";
+import ItemClient from "../../services/taskService";
+import { DragDropContext, Draggable, Droppable } from "react-beautiful-dnd";
+import "./taskList.css";
+
+const TasksList = ({
+  tasks,
+  setTasks,
+  setEditTask,
+  searchInput,
+  statusFilter,
+  setPresentedTasksNum,
+}) => {
+  const [filteredTasks, setFilteredTasks] = useState([]);
+  const taskService = new ItemClient();
+
+  useEffect(() => {
+    const searchFilteredTasks = tasks.filter((task) => {
+      return task.itemName.toLowerCase().includes(searchInput.toLowerCase());
+    });
+
+    let statusFilteredTasks = [];
+
+    switch (statusFilter) {
+      case "completed":
+        statusFilteredTasks = searchFilteredTasks.filter(
+          (task) => task.status === true
+        );
+        break;
+      case "uncompleted":
+        statusFilteredTasks = searchFilteredTasks.filter(
+          (task) => task.status === false
+        );
+        break;
+      default:
+        statusFilteredTasks = searchFilteredTasks;
+        break;
+    }
+
+    setFilteredTasks(statusFilteredTasks);
+    setPresentedTasksNum(statusFilteredTasks.length);
+  }, [tasks, searchInput, statusFilter]);
+
+  const handleDelete = async (task) => {
+    setTasks(tasks.filter((t) => t.id !== task.id));
+    await taskService.removeTask(task.id);
```

Comment:
what if this request failed? you delete the item from the DOM even though it wasn't really got deleted.